src/main/java/net/minecraftforge/common/BiomeManager.java

```diff
@@ -123,6 +166,83 @@ public static void removeBiome(BiomeType type, BiomeEntry entry)
         return list != null ? ImmutableList.copyOf(list) : null;
     }
 
+    /**
+     * Gets the ocean biome entries for a type of ocean.
+     * This method should be called to get the ocean biome entries for a type of ocean.
+     * This method is threadsafe.
+     *
+     * @param type The type of ocean to get the entries for.
+     * @return An ImmutableList of the ocean biome entries for the ocean type or null if there are no entries.
```

Comment:
This is incorrect, if a biome is added and then removed so that the list becomes empty, an empty list is returned instead of `null` even though there are no entries.
This makes it so your patch code above will fail, because it checks for null specifically.

Reply:
Will fix. 
As said above, I will be making the nullable lists return an empty list instead.